src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -260,18 +262,75 @@ private void resetTuning() {
     numServesBlocking = 0;
   }
 
+  private static final Pattern RANGE = Pattern.compile("^bytes=(\\d+)-(\\d*)$");
+
   private void sendFile(RoutingContext context, String file, FileProps fileProps) {
     HttpServerRequest request = context.request();
+
+    // check if the client is making a range request
+    String range = request.getHeader("Range");
+    // parse it
+    Long offset = null;
+    // end byte is length - 1
+    Long end = fileProps.size() - 1;
+
+    if (range != null) {
+      Matcher m = RANGE.matcher(range);
+      if (m.matches()) {
+        try {
+          String part = m.group(1);
+          // offset cannot be empty
+          offset = Long.parseLong(part);
+          // offset must fall inside the limits of the file
+          if (offset < 0 || offset >= fileProps.size()) {
+            throw new IndexOutOfBoundsException();
+          }
+          // length can be empty
+          part = m.group(2);
+          if (part != null && part.length() > 0) {
+            // ranges are inclusive
+            end = Long.parseLong(part);
+            // offset must fall inside the limits of the file
+            if (end < offset || end >= fileProps.size()) {
+              throw new IndexOutOfBoundsException();
+            }
+          }
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+          context.fail(416);
+          return;
+        }
+      }
+    }
+
     writeCacheHeaders(request, fileProps);
 
+    // notify client we support range requests
+    MultiMap headers = request.response().headers();
+    headers.set("Accept-Ranges", "bytes");
```

Comment:
Actually we do, if we don't send it browsers will not for example resume downloads, or media players will not seek to a specific byte to start streaming form.

If we make it configurable then the whole resumable feature is disable. That is also a option but i don't see why would one not want that for static files. Their content is not dynamic so it does not have any side effects and can help reduce bandwidth for bad connections that break often (think mobile for example).